### PR TITLE
Formalize the notion of a placeholder domain

### PR DESCRIPTION
**Pull request by evancharlton**
Introduce a placeholder domain (owned by Scaler) which will allow for
children to assume a domain exists and is valid, but the Scaler can know
whether or not there is a valid domain or not.

This allows Scaler to use the computed domains and subdomains when
necessary, and use the specified ones when not.

### CODE REVIEW

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-203.surge.sh

**Comment by evancharlton** [comment]
Ping

File: src/components/Scaler/index.js [hunk 1]
```diff
@@ -81,30 +94,36 @@ class Scaler extends Component {
       const updateDomainsForItem = item => {
         domainsByItemId[item.id] = {
           [Axes.time]: this.props.dataContext.timeDomain || [
-            ...Axes.time(
-              this.state.domainsByItemId[item.id] || item.timeDomain
-            ),
+            ...(stripPlaceholderDomain(
```
Comment (valerii-cognite) [inline src/components/Scaler/index.js]: if this one returns `undefined` wouldn't spread operator fail?

Comment (evancharlton) [reply src/components/Scaler/index.js]: Yep. Though now that I look at it, `item.timeDomain` doesn't _necessarily_ exist (it's just that no one has ever used it .. yet). I guess I should add it to `DataProvider` too.

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-203.surge.sh

**Pull request merged**